From bbd8b63fce4455aaf41239ced6e2d956bc2b7499 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Tue, 28 Jul 2020 21:47:58 +0200
Subject: [PATCH] ARROW-9589: [C++/R] Forward declare structs as structs

---
 r/src/arrow_exports.h | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/r/src/arrow_exports.h b/r/src/arrow_exports.h
index 6b8599543..f568d2a14 100644
--- a/r/src/arrow_exports.h
+++ b/r/src/arrow_exports.h
@@ -33,24 +33,24 @@
 namespace arrow {
 
 namespace compute {
-class CastOptions;
+struct CastOptions;
 
 }  // namespace compute
 
 namespace csv {
 
 class TableReader;
-class ConvertOptions;
-class ReadOptions;
-class ParseOptions;
+struct ConvertOptions;
+struct ReadOptions;
+struct ParseOptions;
 
 }  // namespace csv
 
 namespace json {
 
 class TableReader;
-class ReadOptions;
-class ParseOptions;
+struct ReadOptions;
+struct ParseOptions;
 
 }  // namespace json
 
-- 
2.27.0

